### PR TITLE
v3.1 schema update for LMS Connect endpoints

### PR DESCRIPTION
**Pull request by vynmeister**
## Link to JIRA
[SHAPI-996](https://clever.atlassian.net/browse/SHAPI-996)

Review by commit.

Adds new LMS Connect endpoints (Create/Get/Update/Delete Assignment, Get/List/Update Submissions) to the v3.1 schema, as well as an update to the v3.1 `/districts` endpoint adding the `lms_state` field. The LMS Connect endpoints are added to a new `v3.1-lms.yml` file. (To do: updates pending PE feedback)

- Modified files: `full-v3.yml` and `v3/generate.go`
- Files auto-generated/auto-updated: `v3.1.yml`, `v3.1-client.yml`, `v3.1-events.yml`, `v3.1-lms.yml`

## Testing
Ran `make generate VERSION=3` to generate the API schema files for `v3.1` and checked that only v3.1 files were changed.

[SHAPI-996]: https://clever.atlassian.net/browse/SHAPI-996?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: v3/generate.go [hunk 1]
```diff
@@ -141,12 +188,21 @@ func modifyDefinitions(version string, isClient bool, name string, def map[inter
 			home_language_code := properties["home_language_code"].(map[interface{}]interface{})
 			home_language_code["enum"] = languages.ISO6393Codes
 		}
+	case "District":
+		if version == "v3.0" {
+			delete(properties, "lms_state")
+		}
+		if !isClient {
+			if version == "v3.0" {
```
Comment (soto-sandra) [inline v3/generate.go:196]: Same check as above?

File: full-v3.yml [hunk 2]
```diff
@@ -1365,6 +1582,16 @@ definitions:
...
         x-validation: true
+      lms_state:
+        type: string
+        x-nullable: true
+        x-validation: true
+        enum:
+        - "matching_in_progress"
+        - "error"
+        - "disconnected"
+        - ""
```
Comment (soto-sandra) [inline full-v3.yml:1593]: More a curiosity question--in what cases would we get an empty string?

**Review by soto-sandra — APPROVED** [review]
LGTM overall

File: full-v3.yml [hunk 3]
```diff
@@ -1075,6 +1080,212 @@ paths:
...
+      responses:
+        '200':
+          description: OK Response
+          schema:
+            $ref: "#/definitions/AssignmentResponse"
+        '400':
+          $ref: "#/responses/BadRequest"
+        '401':
+          $ref: "#/responses/Unauthorized"
+        '404':
```
[the remark this answers is not in the document]
Comment (vynmeister) [reply full-v3.yml:1109]: Yeah it shouldn't be--I can also modify this in assignments-api

File: full-v3.yml (hunk 2 above)
Comment (vynmeister) [reply full-v3.yml:1593]: They shouldn't get an empty string but I think we've added it as an option in the past just to cover our bases, so just following the pattern here

File: v3/generate.go (hunk 1 above)
Comment (vynmeister) [reply v3/generate.go:196]: Yeah I can refactor this

**Pull request merged**